scertecd/scertecd.go

```diff
@@ -110,13 +112,86 @@ func (s *Server) lazyInit() {
 	})
 }
 
+// updateDynamicDomainList updates Server's knowledge of additional
+// domains, dynamically fetched from setec.
+func (s *Server) updateDynamicDomainList(ctx context.Context) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	sec, err := s.SetecClient.GetIfChanged(ctx, s.Prefix+s.DynamicDomainsKey, s.lastDynVers)
+	if err != nil {
+		switch {
+		case errors.Is(err, api.ErrNotFound):
+			s.Logf("additional domains key %q not present on setec, skipping updating the domains list", s.DynamicDomainsKey)
+			return nil
+		case errors.Is(err, api.ErrValueNotChanged):
+			return nil
+		default:
+			return err
+		}
+	}
```

Comment:
We don't need the `err != nil` check, the switch already covers it.

```suggestion
	switch {
	case errors.Is(err, api.ErrNotFound):
		s.Logf("additional domains key %q not present on setec, skipping updating the domains list", s.DynamicDomainsKey)
		return nil
	case errors.Is(err, api.ErrValueNotChanged):
		return nil
	default:
		return err
	}
```